sdkv2/helper.go

```diff
@@ -390,6 +418,9 @@ func (lb *Helper) queryPlanAPIOption() func(*middleware.Stack) error {
 			middleware.InitializeMiddlewareFunc(
 				queryPlanMiddlewareName,
 				func(ctx context.Context, in middleware.InitializeInput, next middleware.InitializeHandler) (middleware.InitializeOutput, middleware.Metadata, error) {
+					if lb.cfg.KeyRouteAffinity.Type != shared.KeyRouteAffinityNone {
+						lb.queryPlanSeed = lb.getPkHash(in)
+					}
 					return next.HandleInitialize(lb.setQueryPlan(ctx), in)
```

Comment:
You should not change lb.queryPlanSeed it should be set only once at the start of the test, that is all